SCRD-3501 Added addcompute to the replace compute flow

guglo opened the pull request:
This commit includes:
1. Added addcompute to the replace compute flow
2. Moved commit model to the action step when prepare adding compute
3. Added model validation when replace compute. If validation failed, show
the validation errors, remove the new server from model so user can pick
it up when they replace a server again.
4. Fixed defect when user selects an available server without mac-addr and
impi info, the replace button won't be activated
5. Removed obseleted and unused pages.

Note: ignore installOS for now

Commit:
SCRD-3501 Added addcompute to the replace compute flow

This commit includes:
1. Added addcompute to the replace compute flow
2. Moved commit model to the action step when prepare adding compute
3. Added model validation when replace compute. If validation failed, show
  the validation errors, remove the new server from model so user can pick
  it up when they replace a server again.
4. Fixed defect when user selects an available server without mac-addr and
  impi info, the replace button won't be activated
5. Removed obseleted and unused pages.

Note: ignore installOS for now

GarySmith commented on src/pages/UpdateServers.js:
```diff
@@ -447,9 +509,11 @@ class UpdateServers extends BaseUpdateWizardPage {
     return (
       <div className='wizard-page'>
-        <LoadingMask show={this.state.wizardLoading || this.state.loading}/>
+        <LoadingMask show={this.state.wizardLoading || this.state.loading || this.state.validating}
```
wizardLoading should not be in state. It should just use this.props.wizardLoading. The same is true for wizardLoadingErrors. This will also require removing `compnentDidUpdate` (and moving some of that logic into the constructor).

guglo replied:
will reuse existing loading flag. Used it with undefined, '' and 'Validating Changes' to indicate no loading, loading and validating. 
will update compnentDidUpdate

GarySmith commented on src/utils/ModelUtils.js:
```diff
@@ -258,6 +258,15 @@ export function addServerInModel(server, model, props) {
+  return model.updateIn(
+    ['inputModel', 'servers'], list => list.filter(
+      svr => {
+        return svr.get('uid') ? svr.get('uid') !== server.uid : svr.get('id') !== server.id;
```
The first clause here, `svr.get('uid')`, is redundant. You could also use `filterNot` to make the logic a little simpler, i.e.:
`['inputModel', 'servers'], list => list.filterNot(svr => svr.get('uid') === server.uid || svr.get('id') === server.id);`

guglo replied:
will update

guglo commented on src/components/ReplaceServerDetails.js:
```diff
@@ -220,7 +220,18 @@ class ReplaceServerDetails extends Component {
+          if(isComputeNode(this.props.data) && !this.state.isInstallOsSelected) {
+            // if not isInstallOsSelected, mac-addr and ipmpi info can be empty
+            if (REPLACE_SERVER_MAC_IPMI_PROPS.includes(key)) {
+              isValid = isValid.set(key, (valueToCopy.length > 0) || undefined);
```
k

Commit:
Some cleanups:

1. Remove WizardLoading states
2. combine loading and validating
3. combine checkng valid logic in repalce details

Commit:
Simplify code
